components/motor/gpio/motor_encoder.go

```diff
@@ -316,11 +313,20 @@ func (m *EncodedMotor) directionMovingInLock() float64 {
 func (m *EncodedMotor) SetPower(ctx context.Context, powerPct float64, extra map[string]interface{}) error {
 	m.stateMu.Lock()
 	defer m.stateMu.Unlock()
-	return m.setPower(ctx, powerPct, false)
+	if m.rpmMonitorDone != nil {
+		m.rpmMonitorDone()
+	}
+	if m.loop != nil {
+		m.loop.Stop()
+		m.loop = nil
+	}
```

Comment:
[note to self] we cannot use `SetPower `in other parts of the code because `SetPower` will cancel other calls. This means that the other calls have to lock the mutex themselves and call the internal `setPower`